src/genai/_utils/limiters/external_limiter.py

```diff
@@ -20,7 +20,7 @@ class ExternalLimiter(BaseLimiter):
 
     def __init__(self, *, handler: ExternalLimiterHandler):
         self._handler = handler
-        self._lock = asyncio.Lock()
+        self._lock = AsyncioLock()
```

Comment:
I see the problem, nice fix 👍. We could also defer the creation of locks and treat the async objects more carefully, but this is better because you don't have to think about it and we can just drop the fix when we stop supporting `3.9` :)